pydantic/_internal/_generate_schema.py

```diff
@@ -1474,13 +1479,44 @@ def _typed_dict_schema(self, typed_dict_cls: Any, origin: Any) -> core_schema.Co
                         UserWarning,
                     )
 
+                extra_behavior: core_schema.ExtraBehavior = 'ignore'
+                extras_schema: CoreSchema | None = None  # For 'allow', equivalent to `Any` - no validation performed.
+
+                # `__closed__` is `None` when not specified (equivalent to `False`):
+                is_closed = bool(getattr(typed_dict_cls, '__closed__', False))
+                extra_items = getattr(typed_dict_cls, '__extra_items__', typing_extensions.NoExtraItems)
+                if is_closed:
+                    extra_behavior = 'forbid'
+                    extras_schema = None
+                elif not typing_objects.is_noextraitems(extra_items):
+                    extra_behavior = 'allow'
+                    extras_schema = self.generate_schema(replace_types(extra_items, typevars_map))
```

Comment:
Under what conditions does `extra_behavior` remain `'ignore'`?

Reply:
It is first assigned the value of `'ignore'`, so if:
- the TD is not closed
- the TD has no extra items
- the extra from the config is not `'allow'` or `'forbid'`